staging/src/k8s.io/pod-security-admission/api/helpers.go

```diff
@@ -237,16 +241,27 @@ func PolicyToEvaluate(labels map[string]string, defaults Policy) (Policy, field.
 		errs = appendErr(errs, err, AuditVersionLabel, version)
 	}
 	if level, ok := labels[WarnLevelLabel]; ok {
+		hasWarnLevel = true
 		p.Warn.Level, err = ParseLevel(level)
 		errs = appendErr(errs, err, WarnLevelLabel, level)
 		if err != nil {
 			p.Warn.Level = LevelPrivileged // Fail open for warn.
 		}
 	}
 	if version, ok := labels[WarnVersionLabel]; ok {
+		hasWarnVersion = true
 		p.Warn.Version, err = ParseVersion(version)
 		errs = appendErr(errs, err, WarnVersionLabel, version)
 	}
+
+	// Default warn to the enforce level when explicitly set to a more restrictive level.
+	if !hasWarnLevel && hasEnforceLevel && CompareLevels(p.Enforce.Level, p.Warn.Level) > 0 {
+		p.Warn.Level = p.Enforce.Level
+		if !hasWarnVersion {
+			p.Warn.Version = p.Enforce.Version
+		}
```

Comment:
I'm not sure we want to default warning level if they've specified warning version... today, the version labels are inert if you don't specify a level, right?

Reply:
Yeah, I wasn't sure exactly what to do in this case. Today, version levels are still applied to the default level. In most cases this is probably `privileged`, but if you override the defaults then the version would be applied to those.

The behavior is sort of weird in both cases, but I decided that what I have here (apply the `warn-version` to the defaulted level) was the least-unexpected outcome. Picture this:
1. You set `enforce=baseline` on your namespace
2. You now get warnings on deployments that don't meet the baseline standards
3. You set `warn=latest` cluster-wide
4. You no longer get warnings on your baseline-violating deployments?

Reply:
ok, I buy that